### PR TITLE
fix(server, databae): fix bugs on #52.

### PR DESCRIPTION
**Pull request by kehiy**
## Related Issue

- Fixes #(issue number)

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -4,41 +4,37 @@ import (
...
-
 type Database struct {
 	DBName       string
 	QueryTimeout time.Duration
 	Client       *mongo.Client
 }
 
-func New(cfg Config) (*Database, error) {
+func Connect(cfg Config) (*Database, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(cfg.ConnectionTimeout)*time.Millisecond)
```
Comment (ZigBalthazar) [inline database/database.go:19]: Add a new timeout to the configuration to initialize the connection. "connectionTimeout" is used to manage connection pool timeout, not to initialize the connection.

File: database/database.go [hunk 2]
```diff
@@ -4,41 +4,37 @@ import (
...
 
-	client, err := mongo.Connect(context.TODO(), opts)
+	client, err := mongo.Connect(ctx, opts)
 	if err != nil {
 		return nil, err
 	}
 
 	var result bson.M
-	if err := client.Database("admin").RunCommand(context.TODO(), bson.D{{Key: "ping", Value: 1}}).
+	if err := client.Database("admin").RunCommand(ctx, bson.D{{Key: "ping", Value: 1}}).
```
Comment (ZigBalthazar) [inline database/database.go:37]: use `queryTimeout`

File: handler/event.go [hunk 3]
```diff
@@ -3,26 +3,158 @@ package handler
...
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 func (h *Handler) HandleEvent(e *event.Event) error {
-	collName, ok := database.KindToCollectionName[e.Kind]
+	collName, ok := KindToCollectionName[e.Kind]
 	if !ok {
-		return errors.New("invalid kind")
+		return fmt.Errorf("kind %d is not supported", e.Kind)
```
Comment (ZigBalthazar) [inline handler/event.go:17]:
```suggestion
		return fmt.Errorf("kind %d is not supported yet", e.Kind)
```

File: database/config.go [hunk 4]
```diff
@@ -2,7 +2,7 @@ package database
 
 type Config struct {
 	URI               string
-	DBName            string `yml:"db_name"`
-	ConnectionTimeout int16  `yml:"connection_timeout_in_ms"`
-	QueryTimeout      int16  `yml:"query_timeout_in_ms"`
+	DBName            string `yaml:"db_name"`
+	ConnectionTimeout int32  `yaml:"connection_timeout_in_ms"`
```
Comment (ZigBalthazar) [inline database/config.go:6]: The recommended data type is int16(max: 32767 ~eq: 32.7s) for the timeout.

**Review by ZigBalthazar — CHANGES_REQUESTED** [review]

File: database/database.go (hunk 2 above)
Comment (kehiy) [reply database/database.go:37]: then we need to create 2 contexts. is this logical?

**Review by ZigBalthazar — APPROVED** [review]

**Pull request merged**